packages/toolpad-app/src/components/Home.tsx

```diff
@@ -40,38 +44,55 @@ export interface CreateAppDialogProps {
 function CreateAppDialog({ onClose, ...props }: CreateAppDialogProps) {
   const [name, setName] = React.useState('');
   const createAppMutation = client.useMutation('createApp');
+  const [showAppNameErrorSnackbar, setShowNameErrorSnackbar] = React.useState<boolean>(false);
 
-  return (
-    <Dialog {...props} onClose={onClose}>
-      <DialogForm
-        onSubmit={async (event) => {
-          event.preventDefault();
+  const handleSnackbarClose = React.useCallback(() => setShowNameErrorSnackbar(false), []);
 
-          const app = await createAppMutation.mutateAsync([name]);
-          window.location.href = `/_toolpad/app/${app.id}/editor`;
-        }}
+  return (
+    <React.Fragment>
+      <Dialog {...props} onClose={onClose}>
+        <DialogForm
+          onSubmit={async (event) => {
+            event.preventDefault();
+            try {
+              const app = await createAppMutation.mutateAsync([name]);
+              window.location.href = `/_toolpad/app/${app.id}/editor`;
+            } catch (err) {
+              setShowNameErrorSnackbar(true);
+            }
+          }}
+        >
+          <DialogTitle>Create a new MUI Toolpad App</DialogTitle>
+          <DialogContent>
+            <TextField
+              sx={{ my: 1 }}
+              autoFocus
+              fullWidth
+              label="name"
+              value={name}
+              onChange={(event) => setName(event.target.value)}
+            />
+          </DialogContent>
+          <DialogActions>
+            <Button color="inherit" variant="text" onClick={onClose}>
+              Cancel
+            </Button>
+            <LoadingButton type="submit" loading={createAppMutation.isLoading} disabled={!name}>
+              Create
+            </LoadingButton>
+          </DialogActions>
+        </DialogForm>
+      </Dialog>
+      <Snackbar
```

Comment:
We will only use snackbars when errors happen without context. In this case we can show loading state and error state inline in the dialog, no snackbar needed.